### PR TITLE
update readme to make it more clear, add athena 101 for new comers

### PR DESCRIPTION
**Pull request by Some-random**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,15 +21,17 @@ All of our models are implemented in Tensorflow>=2.0.1. For ease of use, we prov
...
-  - [4) Data Preparation](#4-data-preparation)
-    - [4.1) Create Manifest](#41-create-manifest)
-  - [5) Training](#5-training)
-    - [5.1) Setting the Configuration File](#51-setting-the-configuration-file)
-    - [5.2) Train a Model](#52-train-a-model)
-  - [6) Deployment](#6-deployment)
-  - [7) Results](#7-results)
-    - [7.1) ASR](#71-asr)
-  - [8) Directory Structure](#8-directory-structure)
+  - [4) Training](#5-training)
```
Comment (neneluo) [inline README.md]: the hyperlinks seems wrong

File: README.md [hunk 2]
```diff
@@ -127,99 +134,137 @@ wav_filename	wav_length_ms	transcript
...
-<details><summary>expand json</summary><div>
-
 ```json
 {
-  "batch_size":32,
+  "batch_size":16,
   "num_epochs":20,
-  "sorta_epoch":1,
-  "ckpt":"examples/asr/hkust/ckpts/transformer",
+  "sorta_epoch":1,  # keep batches sorted for sorta_epoch, this helps with the convergence of models
```
Comment (neneluo) [inline README.md:149]: Why some of the comments are highlighted?

File: README.md [hunk 3]
```diff
@@ -127,99 +134,137 @@ wav_filename	wav_length_ms	transcript
...
+  "ckpt":"examples/asr/timit/ckpts/mtl_transformer_ctc_sp/",
+  "summary_dir":"examples/asr/timit/ckpts/mtl_transformer_ctc_sp/event",
 
-  "solver_gpu":[0],
+  "solver_gpu":[2],
```
Comment (neneluo) [inline README.md]: better use gpu [0]

File: README.md [hunk 4]
```diff
@@ -127,99 +134,137 @@ wav_filename	wav_length_ms	transcript
...
-    "data_csv": "examples/asr/hkust/data/train.csv",
-    "audio_config":{"type":"Fbank", "filterbank_channel_count":40},
-    "cmvn_file":"examples/asr/hkust/data/cmvn",
-    "text_config": {"type":"vocab", "model":"examples/asr/hkust/data/vocab"},
-    "input_length_range":[10, 8000]
+    "data_csv": "examples/asr/timit/data/train.csv",
+    "audio_config":{"type":"Fbank", "filterbank_channel_count":40},  # config for feature extraction
+    "cmvn_file":"examples/asr/timit/data/cmvn",  # mean and variance of FBank
+    "text_config": {"type":"eng_vocab", "model":"examples/asr/timit/data/vocab"},  # vocab list
+    "speed_permutation": [0.9, 1.0, 1.1],  # use speed perbutation to increase data diversitty
```
Comment (neneluo) [inline README.md]: "perbutation" -> "perturbation"?

File: README.md [hunk 5]
```diff
@@ -127,99 +134,137 @@ wav_filename	wav_length_ms	transcript
...
-</div></details>
+To get state-of-the-arts models, we usually need to train for more epochs and use ctc joint decoding with language model. These are omitted for to make this tutorial easier to understand.
 
-### 5.2) Train a Model
+### 4.3) Data normalization
+Data normalization is important for the convergence of neural network models. With the generated csv file, we will compute the cmvn file like this 
+```
+python athena/cmvn_main.py examples/asr/$dataset_name/configs/mpc.json examples/asr/$dataset_name/data/all.csv
+```
+The generated cmvn files will be found at ```examples/asr/timit/datta/cmvn```.
```
Comment (neneluo) [inline README.md]: data

File: README.md [hunk 6]
```diff
@@ -127,99 +134,137 @@ wav_filename	wav_length_ms	transcript
...
-    "cmvn_file":"examples/asr/hkust/data/cmvn",
-    "text_config": {"type":"vocab", "model":"examples/asr/hkust/data/vocab"}
+    "cmvn_file":"examples/asr/timit/data/cmvn",
+    "text_config": {"type":"eng_vocab", "model":"examples/asr/timit/data/vocab"}
   }
 }
 ```
 
-</div></details>
+To get state-of-the-arts models, we usually need to train for more epochs and use ctc joint decoding with language model. These are omitted for to make this tutorial easier to understand.
```
Comment (neneluo) [inline README.md]: state-of-the-art

File: README.md [hunk 7]
```diff
@@ -21,17 +21,17 @@ All of our models are implemented in Tensorflow>=2.0.1. For ease of use, we prov
...
+  - [4) Training](#4-training)
+    - [4.1) Prepare the data](#41-prepare-the-data)
+    - [4.2) Setting the Configuration File](#42-settting-the-configuration-file)
+    - [4.3) Data normalization](#43-data-normalization)
+    - [4.4) Train a model](#44-train-a-model)
+    - [4.5) Evaluate a model](#45-evaluate-a-model)
+    - [4.6) Scoring](#46-scoring)
+  - [5) Deployment](#5-deployment)
+  - [6) Results](#6-results)
+    - [6.1) ASR](#611-asr)
```
Comment (neneluo) [inline README.md]: 61

File: README.md [hunk 8]
```diff
@@ -260,7 +260,7 @@ For scoring, you will need to install [sclite](https://github.com/usnistgov/SCTK
 | Median |  1.0   36.0 | 85.0   10.8    2.9    2.8   17.5  100.0 |
 |----------------------------------------------------------------|
 ```
-
+The line with ```Sum/Avg``` is usually what you should be looking for if you just want an overall WER result. In this case, 11.4 is the substitution error, 4.3 is the deletion error, 3.2 is the insertion error and 18.8 is the total WER.
```
Comment (neneluo) [inline README.md]: PER

File: README.md [hunk 9]
```diff
@@ -113,12 +115,17 @@ horovodrun -np 4 -H localhost:4 athena/horovod_main.py examples/translate/spa-en
...
+The data for TIMIT can be found [here](https://www.kaggle.com/mfekadu/darpa-timit-acousticphonetic-continuous-speech) or [here](https://ndownloader.figshare.com/files/10256148). First, we need to download the data and place it at ```examples/asr/timit/data/TIMIT```.
+Then we will run the following scripts, which will do some data precessing and generate data csv for training and development set of TIMIT.
+```
+mkdir -p examples/asr/timit/data
+python examples/asr/timit/local/prepare_data.py examples/asr/timit/data/TIMIT examples/asr/timit/data
+```
+Below is an example csv we generated, it contains the absolute path of input audio, its length and its transcript
 ```csv
 wav_filename	wav_length_ms	transcript
 /dataset/train-clean-100-wav/374-180298-0000.wav	465004	chapter sixteen i might have told you of the beginning of this liaison in a few lines but i wanted you to see every step by which we came i to agree to whatever marguerite wished
```
Comment (neneluo) [inline README.md]: should be timit's csv, and col "speaker" should be included

File: README.md [hunk 10]
```diff
@@ -113,12 +115,17 @@ horovodrun -np 4 -H localhost:4 athena/horovod_main.py examples/translate/spa-en
...
-
-### 4.1) Create Manifest
-
-Athena accepts a textual manifest file as data set interface, which describes speech data set in csv format. In such file, each line contains necessary meta data (e.g. key, audio path, transcription) of a speech audio. For custom data, such manifest file needs to be prepared first. An example is shown as follows:
+## 4) Training
+We will use ASR task TIMIT as an example to walk you through the whole training process. The recipe for this tutorial can be found at ```examples/asr/timit/run_101.sh```.
 
+### 4.1) Prepare the data
+The data for TIMIT can be found [here](https://www.kaggle.com/mfekadu/darpa-timit-acousticphonetic-continuous-speech) or [here](https://ndownloader.figshare.com/files/10256148). First, we need to download the data and place it at ```examples/asr/timit/data/TIMIT```.
+Then we will run the following scripts, which will do some data precessing and generate data csv for training and development set of TIMIT.
```
Comment (neneluo) [inline README.md]: .. data csvs for training, development and test set ..

**Review by neneluo — APPROVED** [review]
lgtm
